### PR TITLE
move listings page into vue

### PR DESCRIPTION
**Pull request by h-m-m**
## Description

This is an initial go at putting the listings page in Vue. I forget who recommend it this past weekend, but the gist is that the listing page makes more sense as a first place to get Vue up and working than the listing form for creating & editing listings. This is my skeleton for that page. Some things are missing, as might be obvious in the screenshot below. I'll also make a note of them in comments further down

## Screenshots

Screen Shot 2020-04-14 at 12 00 51 PM
Screen Shot 2020-04-14 at 12 00 57 PM

### CODE REVIEW

File: app/assets/stylesheets/listings.scss [hunk 1]
```diff
@@ -27,3 +27,13 @@
...
   }
 }
+
+.listing-body table {
+  border-collapse: collapse;
+  tr td, tr th {
+    border-width: 1px;
+    border-color: black;
+    border-style: solid;
+    padding: 1em;
```
Comment (h-m-m) [inline app/assets/stylesheets/listings.scss:37]: Threw this in as a placeholder for later

File: app/javascript/components/PublicListings.vue [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          <th>Created At</th>
+          <th>Name</th>
+          <th>Location</th>
+          <th>Details</th>
+        </tr>
+        <tr v-for="listing in listings" :key="listing.id">
+          <td>{{ listing.created_at }}</td>
+          <td>{{ listing.name }}</td>
+          <td></td>
+          <td></td>
```
Comment (h-m-m) [inline app/javascript/components/PublicListings.vue:15]: I have not populated all these fields yet. Do we have a mock-up of what the important fields are?

File: app/javascript/components/PublicListings.vue [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    var _this = this
+    fetch('/listings.json')
+      .then( result => result.json())
+      .then( json => {
+        _this.listings = json.listings
+        _this.urls = json.urls
+        _this.errors = []
+      })
+      .catch( error => {
+        _this.errors.push(error)
```
Comment (h-m-m) [inline app/javascript/components/PublicListings.vue:41]: TODO: test more error handling

File: app/views/listings/_listing.json.jbuilder [hunk 4]
```diff
@@ -1,2 +1,4 @@
-json.extract! listing, :id, :type, :created_at, :updated_at
+json.extract! listing, :id, :name, :type
+json.created_at "#{listing.created_at.to_formatted_s(:long)} #{listing.created_at.zone}"
+json.updated_at "#{listing.updated_at.to_formatted_s(:long)} #{listing.updated_at.zone}"
```
Comment (h-m-m) [inline app/views/listings/_listing.json.jbuilder:3]: I guess now we have to have a time zone conversation. Should listings show up in time local to the request or to the viewer?

File: app/javascript/components/PublicListings.vue (hunk 2 above)
Comment (maebeale) [reply app/javascript/components/PublicListings.vue:15]: We don't! But this is a great start. I think right now the only thing coming in from our current form is tags. I have some suggestions for additional fields. Best here or in our miro map?

File: app/views/listings/_listing.json.jbuilder (hunk 4 above)
Comment (maebeale) [reply app/views/listings/_listing.json.jbuilder:3]: LOVE that you thought of this! I would vote local to the viewer, but very open to whatever anyone else thinks.

File: Gemfile [hunk 5]
```diff
@@ -47,6 +47,7 @@ group :development, :test do
   gem 'binding_of_caller'
   gem 'factory_bot_rails'
   gem 'pry'
+  gem 'pry-byebug'
```
Comment (maebeale) [inline Gemfile:50]: yay!

**Pull request merged**